bin/prettier.js

```diff
@@ -76,19 +79,55 @@ if (stdin) {
         output = format(input);
       } catch (e) {
         process.exitCode = 2;
-        console.error(e);
+        if (check) {
+          console.log(filename);
+        }
+        console.log(e);
         return;
       }
 
+      if (check) {
+        let output2;
+        try {
+          output2 = format(output);
```

Comment:
In the Jest tests we have the capability to run an AST comparison to make sure that the AST of the pretty-printed code is exactly the same as the original AST. I'm trying to remember what the advantages of that was over this check. Can you think of any advantages of that? Or did I just not realize this check would be better? I think this is nice because you get nicer diffs.

If we want to go this route, do you think we should change the `RUN_AST_TESTS` option in the jest tests to be similar? Instead of comparing ASTs, re-print the output and compare that. I suppose it would make the tests slower because there's an extra printing pass.